src/main/java/org/micks/javasamples/set_samples/WordRepository.java

```diff
@@ -0,0 +1,53 @@
+package org.micks.javasamples.set_samples;
+
+import org.jetbrains.annotations.NotNull;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+
+public class WordRepository {
+
+    private final Set<@NotNull String> words = new HashSet<>();
```

Comment:
First time I see @NotNull in such place inside <String>. I think it is not necessary?